### ISSUE TITLE
Feature: Optional name for activity descriptions

### ISSUE THREAD

**Issue by carlaKC** [bitcoin-dev-project/sim-ln#127]
Add an optional "name" field on `activity` descriptions that can be used as a logging prefix so that it's easy to relate logs to their corresponding activity description.

**Comment by Anyitechs** [comment]
Hi @carlaKC, I'll like to work on this if it's still available.

**Comment by carlaKC** [comment]
Sure! Since this is a user-facing issue, please post a brief design proposal on how you'd like to update the simulation file:
* Changes we'd make to the json to add to the label for actvities
* What we're going to do when we have no labels (use index in json?)
* How we'll handle validation: do we allow some activities with labels and some without?

**Comment by Anyitechs** [comment]
Great! Thank you. I'll come up with a design proposal and share with you here.

**Comment by ikeogu** [comment]
hello everyone, this is great, but can I come up with a design proposal to work on this? I want to contribute.

**Comment by Anyitechs** [comment]
> hello everyone, this is great, but can I come up with a design proposal to work on this? I want to contribute.

I'm working on the proposal already. Had some blockers setting up project locally earlier but fixed that already.

**Comment by ikeogu** [comment]
ohh!, its alright!

**Comment by ikeogu** [comment]
Let me look out for another issue to contribute to.

**Comment by Anyitechs** [comment]
> Sure! Since this is a user-facing issue, please post a brief design proposal on how you'd like to update the simulation file:

Thank you. I'm sorry I didn't come back to this immediately. I had to look at the test cases and made some draft changes to understand the moving parts first.

> * Changes we'd make to the json to add to the label for actvities
We'll add a new field to the activity description. I'm thinking we can label it `activity_name` or just `name`. Then update the following:

- `ActivityParser` struct
- `ActivityDefinition` struct
- `validated_activities`
- Test cases and lastly,
- Use the `activity` "name" on logging.

Let me know if I'm missing something.

> * What we're going to do when we have no labels (use index in json?)
We can use the `source` and `destination` node aliases together as the label if the aliases were provided in json and not the node ID. Otherwise we use the index. For example, Alice to Bob activity can be  "Alice->Bob" as the label, or "Activity 0" as the label if the node ID was used.

> * How we'll handle validation: do we allow some activities with labels and some without?
I'm not sure what the best approach is. I'm thinking we can allow activities with or without labels, but if an activity doesn't have a label, then we can make up a label from their alias or activity index in json. 

Let me know what your thoughts are. @carlaKC

**Comment by carlaKC** [comment]
> We can use the source and destination node aliases together as the label if the aliases were provided in json and not the node ID. Otherwise we use the index. For example, Alice to Bob activity can be "Alice->Bob" as the label, or "Activity 0" (where "0" is the index of the activity) as the label if the node ID was used.

I don't think that we should have multiple ways of doing this depending on what's there, I think it'll be confusing. Would be happy to just use the indexes to start with. We already log the "Alice->Bob" info as part of the description, so I think that way we won't be duplicating anything. 

>  I'm thinking we can allow activities with or without labels, but if an activity doesn't have a label, then we can make up a label from their alias or activity index in json.

SGTM!

Feel free to go ahead and open up a PR, please do open it up early/in draft for a concept ACK so that we can get an early idea of what it'll look like!

**Comment by Anyitechs** [comment]
Hi @carlaKC , I opened a PR draft [here](https://github.com/bitcoin-dev-project/sim-ln/pull/170). Let me know what your thoughts are when you get to take a look at it.

**Comment by alexzaitsev** [comment]
this is next on my radar

**Comment by Onyekachukwu-Nweke** [comment]
@alexzaitsev are you still working on this.

**Comment by alexzaitsev** [comment]
@Onyekachukwu-Nweke I haven't started yet, take it if you want.

**Comment by Onyekachukwu-Nweke** [comment]
> [@Onyekachukwu-Nweke](https://github.com/Onyekachukwu-Nweke) I haven't started yet, take it if you want.

We can collaborate to get it done.

**Comment by Onyekachukwu-Nweke** [comment]
@carlaKC this is the design proposal I came up with.

**Comment by Onyekachukwu-Nweke** [comment]
> # Design Proposal: Optional Activity Names
> ## JSON Schema Changes
> We need to modify the activity descriptions in the simulation file to include an optional "name" field:
> 
> "activities": [
>   {
>     "name": "high_volume_payments",  // New optional field
>     "type": "defined",
>     "source": "node1",
>     "destination": "node2",
>     "start": 60,
>     "count": 100,
>     "wait": 30,
>     "amount": 1000
>   },
>   {
>     // Activity without a name will use auto-generated identifier
>     "type": "random",
>     "source": "node3",
>     "multiplier": 2.0,
>     "payment_amount": 5000
>   }
> ]
> ## Implementation Details
> 1. **Activity Struct Changes**:
>    
>    * Add an optional `name` field to both `DefinedPaymentActivity` and `RandomPaymentActivity` structs
>    * Update the constructors to accept this optional parameter
>    * Modify the `Display` implementation to include the name when available
> 2. **Default Names**:
>    
>    * When no name is provided, generate a name based on the activity index: `activity-{index}`
>    * In logs, use this format: `[{name}] {activity_description}`
> 3. **Validation Rules**:
>    
>    * Names are optional, mixing named and unnamed activities is allowed
>    * If a name is provided, validate that it's a non-empty string
>    * Considering validation for names uniqueness within the simulation to avoid confusion in logs
> 
> ## Code Changes Required
> I'll need to modify the following files:
> 
> 1. `simln-lib/src/defined_activity.rs`
> 2. `simln-lib/src/random_activity.rs`
> 3. The simulation config parsing code to read the new field
> 
> ## Implementation Plan
> 1. Update the structs to include the optional name field
> 2. Modify constructors and display implementations
> 3. Update the config parsing code
> 4. Add validation logic
> 5. Update logging to use the activity name as a prefix

I am revising this design

**Comment by Onyekachukwu-Nweke** [comment]
# Implementation Details

- Add optional `name` field to `ActivityParser` and `ActivityDefinition`
- Generate default names like "Activity-0" for unnamed activities
- Include activity name as prefix in all relevant log messages
- Ensure names are unique across all activities

## JSON Example

```json
{
  "activity": [
    {
      "name": "High-Value-Payments",
      "source": "node1",
      "destination": "node2",
      "interval_secs": 30,
      "amount_msat": 500000
    },
    {
      // No name - will get auto-generated name "Activity-1"
      "source": "node3",
      "destination": "node4",
      "interval_secs": [10, 60],
      "amount_msat": [100000, 300000]
    }
  ]
}
```

Let me know what your thoughts are. @carlaKC

**Comment by carlaKC** [comment]
@Onyekachukwu-Nweke do you still intend to work on this in #253? 
If not I'm going to un-assign the PR so that somebody else can take a shot at it.

**Comment by Onyekachukwu-Nweke** [comment]
I am still working on it, sorry for the radio silence on my part.

**Comment by HarcourtHamsa** [comment]
@carlaKC I'm very new to the sim-ln project. Can I pick this up?

Here's my proposed structure

`"activity": [
    {
      "name": "Some name here" // optional field
      "source": "Alice",
      "destination": "02d804ad31429c8cc29e20ec43b4129553eb97623801e534ab5a66cdcd2149dbed",
      "interval_secs": 1,
      "amount_msat": 2000
    },
  ]`

For activities without `name`, it will be auto generated.

**Comment by carlaKC** [comment]
Fine for you to pick this up, but please note that I'm going to be OOO until 10 Sept so will not ber available for review.

I'd encourage you to take a look at previous unsuccessful attempts to resovle this issue to help inform your implementation - we've already sunk quite a lot of review into what's a fairly trivial feature addition here, so it's important that we learn from past attempts.
